browser/javascript/react-app/src/ampli/index.js

```diff
@@ -331,6 +348,46 @@ export class Ampli {
     this.amplitude?.setGroup(name, value);
   }
 
+  /**
+   * Identify a group and set or update that group's properties.
+   *
+   * @param {string} groupType The group type.
+   * @param {string|string[]} groupName The group name.
+   * @param {Object} properties The group's properties.
+   * @param {string} [properties.optionalString] Description for group optionalString
+   * @param {boolean} properties.requiredBoolean Description for group requiredBoolean
+   * @param {GroupOptions} [options] Options for this groupIdentify call.
+   * @param {MiddlewareExtra} [extra] Extra untyped parameters for use in middleware.
+   */
+  groupIdentify(groupType, groupName, properties, options, extra) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    const event = {
+      event_type: SpecialEventType.Group,
+      event_properties: new Group(properties).event_properties,
```

Comment:
I don't get the purpose here. Why create a Group event, pass in properties, then read properties? I see this pattern is happening in 4 runtimes and also Identify. However, I don't get the benefit for it.

Reply:
@qingzhuozhen Actually it is required only if the event has some const properties. For example:
```
export class Identify implements BaseEvent {
  event_type = 'Identify';
  event_properties: IdentifyProperties & {
    'test_property_name_const_number': 123.4;
  };

  constructor(
    event_properties: IdentifyProperties,
  ) {
    this.event_properties = {
      ...event_properties,
      'test_property_name_const_number': 123.4,
    };
  }
}
```
I can add new condition to codegen templates to use simple logic if const properties do not exist.

Reply:
That makes more sense to me now. Yeah, if we can use a simple logic to determine that would be better. If it is too hard we can also leave it.

Reply:
Done